### PR TITLE
chore(deps): Upgrade Zod to @latest

### LINKED ISSUE #8938: astro@3.4.0 and @astrojs/sitemap@3.0.2 using vulnerable version of Zod
### Astro Info

```block
Astro                    v3.4.0
Node                     v18.18.0
System                   Linux (x64)
Package Manager          npm
Output                   static
Adapter                  none
Integrations             @astrojs/mdx
                         @astrojs/sitemap
```

### If this issue only occurs in one browser, which browser is a problem?

_No response_

### Describe the Bug

Running `npm audit` on a new Astro project raises the [following vulnerability](https://github.com/advisories/GHSA-m95q-7qp3-xv42) with Zod v3.22.1, currently used by both astro@3.4.0 and @astrojs/sitemap@3.0.2.

The vulnerability in Zod is fixed after [v3.22.3](https://github.com/colinhacks/zod/releases/tag/v3.22.3) (latest version is v3.22.4).

### What's the expected result?

Running `npm audit` should not find any security vulnerabilities within Astro.

### Link to Minimal Reproducible Example

https://stackblitz.com/edit/withastro-astro-jqpxng?file=package.json

### Participation

- [ ] I am willing to submit a pull request for this issue.

### PR DESCRIPTION
**Pull request by evadecker**
## Changes

- A security vulnerability was [reported](https://github.com/advisories/GHSA-m95q-7qp3-xv42) in Zod `<= 3.22.2` and a patch was published in Zod `3.22.3`
- This upgrades Zod to `3.22.4` to resolve the dependency vulnerability
- Fixes #8938 

## Testing

- No tests added

## Docs

/cc @withastro/maintainers-docs for feedback!

### CODE REVIEW

**Review by natemoo-re — APPROVED** [review]
Thank you for the contribution!

**Comment by natemoo-re** [comment]
Not sure what's going on with the CI failure, but it's definitely not your fault! We'll debug that.

File: pnpm-lock.yaml [hunk 1]
```diff
@@ -16156,8 +16156,8 @@ packages:
   /setprototypeof@1.2.0:
     resolution: {integrity: sha512-E5LDX7Wrp85Kil5bhZv46j8jOeboKq5JMmYM3gVGdGH8xFpPWXUMsNrlODCrkoxMEeNi/XZIwuRvY4XNwYMJpw==}
 
-  /sharp@0.32.5:
-    resolution: {integrity: sha512-0dap3iysgDkNaPOaOL4X/0akdu0ma62GcdC2NBQ+93eqpePdDdr2/LM0sFdDSMmN7yS+odyZtPsb7tx/cYBKnQ==}
+  /sharp@0.32.6:
+    resolution: {integrity: sha512-KyLTWwgcR9Oe4d9HwCwNM2l7+J0dUQwn/yf7S0EnTtb0eVS4RxO0eUSvxPtzT4F3SY+C4K6fqdv/DO27sJ/v/w==}
```
Comment (evadecker) [inline pnpm-lock.yaml:16160]: @natemoo-re Out of curiosity, do you know why this changed? It seems unrelated to zod.

**Comment by bluwy** [comment]
Zod is specifically pinned because of that types issue. I spent quite some time on it before and it seems there's no way to workaround it (`tsc` just bails). The only way if we really want it to work is to unwrap this as a standalone type i think: https://github.com/withastro/astro/blob/0ab6bad7dffd413c975ab00e545f8bc150f6a92f/packages/astro/src/core/config/schema.ts#L282

**Comment by evadecker** [comment]
@bluwy Thanks for that, I tried digging a little bit, too, but this level of TypeScript is a bit beyond me. Do you know why the pinned version of `zod` works but later releases don't?

**Comment by bumu** [comment]
@evadecker  can you help to fix the conflicts?

**Comment by evadecker** [comment]
> @evadecker can you help to fix the conflicts?

I reran `pnpm i` to get rid of the lockfile merge conflict, but we still have the issue that @bluwy mentioned.

File: .changeset/swift-rivers-impress.md [hunk 2]
```diff
@@ -0,0 +1,7 @@
+---
+'@astrojs/markdoc': patch
+'@astrojs/sitemap': patch
+'astro': patch
+---
+
+Upgrade zod to 3.22.4
```
Comment (sarah11918) [inline .changeset/swift-rivers-impress.md:7]:
```suggestion
Upgrades Zod to 3.22.4
```

**Review by sarah11918 — APPROVED** [review]
Just updating the verb tense of the changeset for consistency! Also, when just using the name of the thing, Zod appears to be capitalized, so did that too.  

Docs will be happy with these changes! :raised_hands:

**Comment by evadecker** [comment]
@bluwy @natemoo-re 

I was able to resolve the previous error...

```
Error: src/core/config/schema.ts(62,14): error TS2742: The inferred type of 'AstroConfigSchema' cannot be named without a reference to '.pnpm/mdast-util-to-hast@12.3.0/node_modules/mdast-util-to-hast'. This is likely not portable. A type annotation is necessary.
Error: src/core/config/schema.ts(312,17): error TS2742: The inferred type of 'createRelativeSchema' cannot be named without a reference to '.pnpm/mdast-util-to-hast@12.3.0/node_modules/mdast-util-to-hast'. This is likely not portable. A type annotation is necessary.
```

...without unwrapping `AstroConfigType` as a standalone type. 🎉 

To do this required explicitly adding `mdast-util-to-hast` as a devDependency, and importing `mdast-util-to-hast` in the config file. The extra `import` feels odd to me, but it's the [solution suggested by the TypeScript team developer lead](https://github.com/microsoft/TypeScript/issues/29808#issuecomment-1320713018) when TypeScript's declaration emitter fails to resolve a reference.

**Comment by natemoo-re** [comment]
@evadecker wow great job! I think that works. I'm curious if we'd need to add it as an actual `dependency` to ensure it's included in user's projects, though.

**Comment by evadecker** [comment]
Happy to make an edit if that's the case!

**Review by ematipico — APPROVED** [review]
@bluwy could you please take a look?

**Review by bluwy — COMMENTED** [review]
One small request, but great work on fixing this! Looks like the move from shiki to shikiji allowed an easier fix here.

**Review by natemoo-re — APPROVED** [review]
Thanks so much for the PR @evadecker!
